charmhelpers/contrib/openstack/templating.py

```diff
@@ -218,13 +235,18 @@ def __init__(self, templates_dir, openstack_release):
             else:
                 apt_install('python3-jinja2')
 
-    def register(self, config_file, contexts):
+    def register(self, config_file, contexts, config_template=None):
         """
         Register a config file with a list of context generators to be called
         during rendering.
+        config_template can be used to load a template from a string instead of
+        using template loaders and template files.
```

Comment:
Opportunity here to add params and their types to the docstring.  Partiularly for the `config_template` param which is actually an string that gets converted to an OSConfigTemplate.

Reply:
ack